pkg/controllers/user/pipeline/hooks/drivers/github.go

```diff
@@ -69,14 +69,29 @@ func (g GithubDriver) Execute(req *http.Request) (int, error) {
 	}
 
 	ref, branch, commit := "", "", ""
+	optionalEnvVars := map[string]string{}
```

Comment:
Why we need optionalEnvVars to store tag? what is the different between tag and commit?

Reply:
Tag is not included in pipeline execution fields while commit is.